kivy/uix/modalview.py

```diff
@@ -285,32 +270,38 @@ def on_touch_up(self, touch):
         self._touch_started_inside = None
         return True
 
-    def on__anim_alpha(self, instance, value):
-        if value == 0 and self._window is not None:
+    def on__anim_alpha(self, _instance, value):
+        """ animation progress callback. """
+        if value == 0 and self._is_open:
             self._real_remove_widget()
 
     def _real_remove_widget(self):
-        if self._window is None:
+        if not self._is_open:
             return
-        self._window.remove_widget(self)
-        self._window.unbind(
+        Window.remove_widget(self)
+        Window.unbind(
```

Comment:
While technically your change is ok, I would feel better if you leave this unchanged. Because although `self._window` is `Window`, it is possible for the Window to have changed, e.g. during testing when we may create a new window for each test (although currently we actually re-use it). 

Also, on principle, it's better to remove/unbind from from `self._window` because we know that is the original object we use for sure, while Window, although very likely to also be the same object, ould possibly be changed in the future so it's less robust.